src/tests/ftest/server/multiengine_persocket.py

```diff
@@ -159,9 +158,25 @@ def host_reboot(self, hosts):
         """
         reboot_waittime = self.params.get("reboot_waittime", "/run/server_config/*", default=210)
         cmd = "sudo shutdown -r now"
-        run_pcmd(hosts, cmd, timeout=210)
+        run_remote(self.log, hosts, cmd, timeout=210)
         self.log.info("===Server %s rebooting... \n", hosts)
         time.sleep(reboot_waittime)
+        ping_timeout = 600
+        start = time.time()
+        ping_status = None
+        while ping_status is None and (time.time() - start) < ping_timeout:
+            ping_result = run_local(self.log, ["ping", "-c 1", str(hosts)], check=False).stdout
+            self.log.info("===>ping_result= %s", ping_result)
+            ping_status = re.search("1 received", ping_result)
+            if (time.time() - start) < 120:
+                time.sleep(5)
+            else:
+                time.sleep(1)
+        if time.time() - start >= ping_timeout:
+            self.fail("#{0} reboot failed, did not come back after {1} seconds",
+                      hosts, ping_timeout)
+        time.sleep(5)
```

Comment:
We should not use a sleep here.  Instead, we could clush (`run_remote()`) a basic command, like `uname`, in a while loop until it passes on all the hosts or hits a timeout (to avoid an invite loop).